### PR TITLE
khalid/statement issues

### PR DESCRIPTION
**Pull request by Khalidxx**
Issues fixed:
- Showing barrier of `-` instead of `0` for `Match/Differ` and `Over/Under` contracts
- Showing multiple `rateLimit` error messages when scrolling to the end of Statements window

### CODE REVIEW

File: src/statement/statement.es6 [hunk 1]
```diff
@@ -89,6 +90,7 @@ const refreshTable  = (yyy_mm_dd) => {
             .then(refresh)
             .catch((err) => {
                   refresh({});
+                  api_limit_reached = true;
```
Comment (OskarAhl) [inline src/statement/statement.es6]: Should this be reset after one minute? (since the rate limit will be cleared by then)

File: src/common/rivetsExtra.es6 [hunk 2]
```diff
@@ -67,7 +67,7 @@ rv.formatters['has-value'] = hasValue;
 /* localise price format*/
 rv.formatters['format-price'] = (value, currency) => hasValue(value) ? formatPrice(value, currency) : undefined;
 /* comma added format*/
-rv.formatters['add-comma'] = (value, decimal_points) => (parseFloat(value)) ?  addComma(value, decimal_points) : undefined;
+rv.formatters['add-comma'] = (value, decimal_points) => value ?  addComma(value, decimal_points) : undefined;
```
Comment (OskarAhl) [inline src/common/rivetsExtra.es6]: Since value can be a number, we could add the `hasValue` check here also?

Comment (Khalidxx) [reply src/common/rivetsExtra.es6]: Yeah, that should be better

File: src/viewtransaction/viewTransaction.es6 [hunk 3]
```diff
@@ -233,11 +239,11 @@ const updateIndicative = (data, state) => {
     state.proposal_open_contract.current_spot = contract.current_spot;
     state.proposal_open_contract.current_spot_time = contract.current_spot_time;
     state.proposal_open_contract.bid_price = contract.bid_price;
-    state.proposal_open_contract.entry_tick = contract.entry_tick;
+    state.proposal_open_contract.entry_tick = countDecimals(contract.entry_tick) < 3 ? contract.entry_tick.toFixed(3) : contract.entry_tick;
```
Comment (OskarAhl) [inline src/viewtransaction/viewTransaction.es6]: Why is this necessary?

File: src/viewtransaction/viewTransaction.es6 [hunk 4]
```diff
@@ -45,6 +45,12 @@ const showMarketDataDisruptionWindow = () => {
    window.dd = market_data_disruption_win;
 };
 
+const countDecimals = (value) => { 
+  if ((value % 1) != 0)
```
Comment (OskarAhl) [inline src/viewtransaction/viewTransaction.es6]: Should be `!==`

File: src/viewtransaction/viewTransaction.es6 (hunk 3 above)
Comment (Khalidxx) [reply src/viewtransaction/viewTransaction.es6]: It's because the entry/exit spot doesn't show enough decimal places. For example `123.45` is supposed to be shown `123.450`. Sometimes they choose a `match/differ` contract and last digit prediction as 5, which means they lost in this case, but they see last digit as 5. They will get really angry and complain 😂

File: src/viewtransaction/viewTransaction.es6 (hunk 4 above)
Comment (Khalidxx) [reply src/viewtransaction/viewTransaction.es6]: Yes, for sure

File: src/viewtransaction/viewTransaction.es6 (hunk 3 above)
Comment (OskarAhl) [reply src/viewtransaction/viewTransaction.es6]: 👍 ah yes, I think we can use `display_decimals` instead of `3`?

Comment (Khalidxx) [reply src/viewtransaction/viewTransaction.es6]: Oh okay, I'll change it
